operator/src/main/java/oracle/kubernetes/operator/helpers/ConfigMapHelper.java

```diff
@@ -47,16 +48,92 @@ public ScriptConfigMapStep(String namespace, Step next) {
+              public NextAction onSuccess(Packet packet, V1ConfigMap result, int statusCode,
+                  Map<String, List<String>> responseHeaders) {
+                
+                LOGGER.info(MessageKeys.CM_CREATED, namespace);
+                packet.put(ProcessingConstants.SCRIPT_CONFIG_MAP, result);
+                return doNext(packet);
+              }
+            });
+            return doNext(create, packet);
+          } else if (AnnotationHelper.checkFormatAnnotation(result.getMetadata()) && result.getData().entrySet().containsAll(cm.getData().entrySet())) {
+            // existing config map has correct data
+            LOGGER.fine(MessageKeys.CM_EXISTS, namespace);
+            packet.put(ProcessingConstants.SCRIPT_CONFIG_MAP, result);
+            return doNext(packet);
+          } else {
+            // we need to update the config map
+            Map<String, String> updated = result.getData();
+            updated.putAll(cm.getData());
+            cm.setData(updated);
+            Step replace = factory.create().replaceConfigMapAsync(cm.getMetadata().getName(), namespace, cm, new ResponseStep<V1ConfigMap>(next) {
+              @Override
+              public NextAction onFailure(Packet packet, ApiException e, int statusCode,
+                  Map<String, List<String>> responseHeaders) {
+                return super.onFailure(ScriptConfigMapStep.this, packet, e, statusCode, responseHeaders);
+              }
+              
+              @Override
+              public NextAction onSuccess(Packet packet, V1ConfigMap result, int statusCode,
+                  Map<String, List<String>> responseHeaders) {
+                LOGGER.info(MessageKeys.CM_REPLACED, namespace);
+                packet.put(ProcessingConstants.SCRIPT_CONFIG_MAP, result);
+                return doNext(packet);
+              }
+            });
+            return doNext(replace, packet);
+          }
+        }
+      });
+      
+      return doNext(read, packet);
+    }
+
+    // Make this protected so that it can be unit tested
+    protected V1ConfigMap computeDomainConfigMap() {
       String name = KubernetesConstants.DOMAIN_CONFIG_MAP_NAME;
       V1ConfigMap cm = new V1ConfigMap();
-      
+      cm.setApiVersion("v1");
+      cm.setKind("ConfigMap");
+
       V1ObjectMeta metadata = new V1ObjectMeta();
       metadata.setName(name);
       metadata.setNamespace(namespace);
       
       AnnotationHelper.annotateWithFormat(metadata);
       
       Map<String, String> labels = new HashMap<>();
+      // This config map is a singleton that is shared by all the domains
+      // We need to add a domain uid label so that it can be located as
+      // related to the operator.  However, we don't have a specific domain uid
+      // to set as the value.  So, just set it to an empty string.  That way,
+      // someone seleting on just the weblogic.domainUID label will find it.
+      labels.put(LabelConstants.DOMAINUID_LABEL, "");
```

Comment:
Why is it necessary that this cm be found by a search for domainUID?

Reply:
It's part of the labeling effort - all operator related artifacts should have a
weblogic.operatorName label, and all domain related artifacts should have
weblogic.domainUID label.  That way, a customer can find all of operator / domain
related kubernetes artifacts using a selector that checks for the presence of either
of these labels.

Reply:
For all of the operator artifacts, which are essentially singletons, we set the weblogic.operatorName value to the operator's namespace.  For all of the other domain
 elated artifacts, we set weblogic.domainUID to that domain's uid.  However, since this config
map is shared by all domains, and is still domain related, I needed to add the
weblogic.domainUID tag to indicate that the config map is domain related, but I don't have
a specific domainUID (since all domains used it), so I just set it to an empty string.

This seemed better than inventing yet another tag to indicate that config map was domain related.

(Tom Barnes & I worked out this strategy)